system/shcommon.py

```diff
@@ -177,7 +175,10 @@ def is_binary_file(filename, nbytes=1024):
     """
     with open(filename, 'rb') as ins:
         for c in ins.read(nbytes):
-            oc = ord(c)
+            if isinstance(c, six.integer_types):
+                oc = c
+            else:
+                oc = ord(c)
```

Comment:
oc = c in isinstance(c, int) else ord(c)  # does that not work?

Reply:
@cclauss Sure, it would work, but is there any reason for using compact syntax here? In most cases, it is a bit easier to read full `if`/`for`/`...` statements than to read their compact equivalents.